### ISSUE TITLE
Error: Shared folder name has improper format

### ISSUE THREAD

**Issue by fakeuser999** [lastpass/lastpass-cli#164]
My lastpass-cli was working fine on OSX Mavericks (via MacPorts) until roughly a couple weeks ago.  I looked into it today, and updated my lastpass-cli to the latest in order to fix the certificate upgrade requirement.

I am now able to use lastpass-cli to login and retrieve passwords without issue.  But when I attempt to generate a new password, the tool throws an error:
```

computername:~ username$ lpass generate --sync=now --no-symbols --url=http://10.9.8.7 10.9.8.7 12
Error: Shared folder name has improper format
```

This error seems to come from blob::account_assign_share(), but I don't recall using the shared folder feature.  Just to see what would happen, after discovering this I created my first shared folder, and it didn't have any effect in resolving the issue.  I also tried logging out and logging back in via lastpass-cli.

**Comment by bcopeland** [comment]
Can you try this (untested) patch?
[0001-blob-don-t-exit-in-assign-share-if-no-group.txt](https://github.com/lastpass/lastpass-cli/files/198392/0001-blob-don-t-exit-in-assign-share-if-no-group.txt)

**Comment by fakeuser999** [comment]
The patch worked, thanks.

I had to hand-merge in the patch since your file length is not the same as what comes down from MacPorts.  I patched the MacPorts built binary using the instructions below as a template.  The activate step failed, but I was able to copy the lpass binary from the work directory to /opt/local/bin/ and it can now generate passwords fine.

https://stackoverflow.com/questions/10863698/macports-ncurses-error-when-trying-to-upgrade
